R/req-perform-multi.R

```diff
@@ -0,0 +1,228 @@
+#' Perform a multi request
+#'
+#' Perform a request with a multi policy, e.g. created by [req_paginate()] or
+#' [req_chunk()].
+#'
+#' @inheritParams req_perform
+#' @param max_requests The maximum number of requests to perform.
+#' @param cancel_on_error Should all pending requests be cancelled when you
+#'   hit an error. Set this to `TRUE` to stop all requests as soon as you
+#'   hit an error. Responses that were never performed will have class
+#'   `httr2_cancelled` in the result.
+#' @param progress Display a progress bar? Use `TRUE` to turn on a basic progress
+#'   bar, use a string to give it a name, or see [progress_bars] for more details.
+#'
+#' @return The result of `vec_c()`ing together the `data` fields that were
+#'   extracted by the `parse_resp` argument of [req_paginate()].
+#'   If `parse_resp` is not specified, it will be a list of the raw responses.
+#' @export
+#'
+#' @examples
+#' page_size <- 150
+#'
+#' req_pokemon <- request("https://pokeapi.co/api/v2/pokemon") %>%
+#'   req_url_query(limit = page_size) %>%
+#'   req_paginate_next_url(
+#'     parse_resp = function(resp) {
+#'       parsed <- resp_body_json(resp)
+#'       results <- parsed$results
+#'       data <- data.frame(
+#'         name = sapply(results, `[[`, "name"),
+#'         url = sapply(results, `[[`, "url")
+#'       )
+#'
+#'       list(data = data, next_url = parsed$`next`)
+#'     },
+#'     n_pages = function(parsed) {
+#'       total <- parsed$count
+#'       ceiling(total / page_size)
+#'     }
+#'   )
+#'
+#' responses <- req_perform_multi(req_pokemon)
+req_perform_multi <- function(req,
```

Comment:
We need to think about the naming of functions. There currently is `multi_req_perform()` to perform a list of requests in parallel. It might make sense to rename this to something like `reqs_perform_parallel()` (though I'm not quite sure we want to add the prefix `reqs`).